File: app-flutter/lib/models/product.dart
class Product {
  const Product({
    required this.id,
    required this.title,
    required this.sku,
    required this.description,
    required this.mainImage,
    required this.costPrice,
    required this.desiredMargin,
  });

  factory Product.fromJson(Map<String, dynamic> json) => Product(
        id: (json['id'] as num?)?.toInt() ?? 0,
        title: json['title'] as String? ?? '',
        sku: json['sku'] as String? ?? '',
        description: json['description'] as String? ?? '',
        mainImage: json['main_image'] as String?,
        costPrice: (json['cost_price'] as num?)?.toDouble() ?? 0,
        desiredMargin: (json['desired_margin'] as num?)?.toDouble() ?? 0,
      );

  final int id;
  final String title;
  final String sku;
  final String description;
  final String? mainImage;
  final double costPrice;
  final double desiredMargin;
}

File: app/Models/Product.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Product extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'title',
        'sku',
        'description',
        'main_image',
        'cost_price',
        'desired_margin',
        'attributes',
    ];

    protected $casts = [
        'cost_price' => 'decimal:2',
        'desired_margin' => 'decimal:2',
        'attributes' => 'array',
    ];

    // RELATIONSHIPS
    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function listings()
    {
        return $this->hasMany(Listing::class);
    }
}
